book/visual-effects.md

```diff
@@ -144,14 +216,77 @@ class Pixel:
         self.a = a
 ```
 
-Note that this `Pixel` code is an illustrative example, not something
+If you'll permit me a short digression here: what exactly do "red",
+"green", and "blue" mean here? Formally, they are defined by the [sRGB
+color space][srgb], meant to represent the colors of common computer
+screens.[^lcd-design] Combinations of those three basic colors form a
+_gamut_ of possible colors that the screen can display.[^other-spaces]
+
+[^lcd-design]: Actually, some screens contain [pixels besides red,
+    green, and blue][lcd-design], including white, cyan, or yellow.
+    Moreover, different screens can use slightly different reds,
+    greens, or blues; professional color designers typically have to
+    [calibrate their screen][calibrate] to display colors accurately.
+    For the rest of us, the software still communicates with the
+    display in terms of standard red, green, and blue colors, and the
+    display hardware converts to whatever pixels it uses.
+
+[^other-spaces]: The sRGB color space dates back to [CRT
+    displays][CRT]. New technologies like LCD, LED, and OLED can
+    display more colors, so CSS now includes [syntax][color-spec] for
+    expressing these new colors. All color spaces have a limited gamut
+    of expressible colors.
+    
+[lcd-design]: https://geometrian.com/programming/reference/subpixelzoo/index.php
+[calibrate]: https://en.wikipedia.org/wiki/Color_calibration
+[srgb]: https://en.wikipedia.org/wiki/SRGB
+[CRT]: https://en.wikipedia.org/wiki/Cathode-ray_tube
+[color-spec]: https://drafts.csswg.org/css-color-4/
+
+By the way, this `Pixel` code is an illustrative example, not something
 our browser has to define. It's standing in for somewhat more complex
-code within Skia itself.
+code within Skia itself. What Skia *does* have, though, is a `Color`
+type, which represents the color of an abstract pixel somewhere. Since
+we're already talking about colors, let's implement a helper function
+to construct a Skia `Color` from a string:
 
-Now there's an SDL surface for the window contents and a Skia surface
-that we can draw to. We'll raster\index{raster} to the Skia surface,
-and then once we're done we'll copy it to the SDL surface to
-display on the screen.
+``` {.python replace=r%2c%20g%2c%20b/r%2c%20g%2c%20b%2c%20a}
+def parse_color(color):
```

Comment:
You need to make sure to parse all colors used by server*.py and all demos.

Reply:
I think my plan is to update all colors to be hex.